src/Utils/Misc.idr

```diff
@@ -96,7 +96,7 @@ export
 split_at_concat_rev : (n : Nat) -> (xs : Vect (n + m) a) -> {0 l : Vect n a} -> {0 r : Vect m a} -> (0 _ : splitAt n xs = (l, r)) -> l ++ r = xs
 split_at_concat_rev Z _ Refl = Refl
 split_at_concat_rev (S n) (x::xs) {l} prf with (splitAt n xs) proof sprf
-  split_at_concat_rev (S n) (x::xs) {l=x::l} Refl | (l, r) = cong (x::) $ split_at_concat_rev n xs sprf
```

Comment:
`r` is bound to `Vect` of multiplicity of 0. Here it should be type of vector element.

Reply:
I don't quite get what you mean. I'm sorry... but if I'm not mistaken, I think `(l, r)` corresponds to the result of `splitAt n xs`, which returns the left splice, and the right splice of `xs` when splitted at index `n`. Also, changing `r` to `element` will result in:
```
Pattern variable element unifies with: ?r [no locals in scope].
```